### PR TITLE
updated the password validator

### PR DESCRIPTION
**Pull request by shikharsaurav**
Now password will also check for follwing in signup page
minimum 1 Upper case alphabet
minimum 1 lower case alphabet
minimum 1 numeric character
minimum 1 special character

### CODE REVIEW

File: lib/utils/validator.dart [hunk 1]
```diff
@@ -45,10 +45,14 @@ class Validator {
   // }
 
   static String validatePassword(String value) {
+    String pattern =
+        r'^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#\$&*~]).{8,}$';
+    RegExp regExp = new RegExp(pattern);
     if (value.length < 6) {
       return 'Password must be at least 6 characters.';
+    } else if (!regExp.hasMatch(value)) {
+      return 'Password must atleast 1 A-Z,a-z,0-9&special character';
```
Comment (Sagar2366) [inline lib/utils/validator.dart:54]: User won't understand what is meaning of this.
Please show meaningful message, toast or suggestion.

Comment (shikharsaurav) [reply lib/utils/validator.dart:54]: can u give some suggestion plz @Sagar2366 
it will be really helpful

Comment (shikharsaurav) [reply lib/utils/validator.dart:54]: what should be printed instead?

Comment (Sagar2366) [reply lib/utils/validator.dart:54]: If you can show toast/message box which can clearly show why the error that would be great addition.
Sample : https://ux.stackexchange.com/questions/108307/a-clear-error-message-for-wrong-password-format

Comment (shikharsaurav) [reply lib/utils/validator.dart]: plz see i have updated it if it is not workable i will change it

Comment (shikharsaurav) [reply lib/utils/validator.dart]: thanks for suggestion

Comment (Sagar2366) [reply lib/utils/validator.dart]: @shikharsaurav while entering password only you can show toast, should give idea which criteria is matched and which are still left.
sample - 
Screenshot 1942-12-28 at 6 36 05 PM

Comment (shikharsaurav) [reply lib/utils/validator.dart]: ok sir i will do it like that

**Comment by shikharsaurav** [comment]
i have added plz check it

**Comment by Sagar2366** [comment]
> i have added plz check it

This is good addition. Please resolve conflicts on this PR and make password checker container design look good to match with our current project UI.

**Comment by shikharsaurav** [comment]
Ok sir

**Comment by shikharsaurav** [comment]
Screenshot_1616332158

**Comment by shikharsaurav** [comment]
is it okay sir @Sagar2366 ?

**Comment by Sagar2366** [comment]
@shikharsaurav still need improvement in UI, not looking good.

**Comment by shikharsaurav** [comment]
sry sir ,actually i m new to flutter & open source projects

**Comment by shikharsaurav** [comment]
i will try my best to improve myself

**Comment by shikharsaurav** [comment]
ut

**Comment by shikharsaurav** [comment]
is this UT ok for password validator

**Comment by Sagar2366** [comment]
Screenshot 2021-03-23 at 9 52 43 AM

@shikharsaurav don't worry, all mentors are here to help and guide you.
1. Add UT's whatever you have cerated till now to PR.
2. Pull master changes to resolve Travis CI issue
3. We will merge the PR with current UI you have designed. Just my observation, even all criteria's are passed, password strength button doesn't reach to 100% or show green colour. Can you add those changes and send PR again for review ?

**Comment by shikharsaurav** [comment]
https://user-images.githubusercontent.com/47968768/112093598-bb296a00-8bbf-11eb-817a-e6497647167b.mp4

this sir

**Comment by shikharsaurav** [comment]
@Sagar2366  i have made PR #469 & #471 which is successfully checked by travis , u can merge them then i will close it
thank u

**Pull request closed without merging**